Aws/Core.hs

```diff
@@ -604,10 +605,8 @@ signature :: Credentials -> AuthorizationHash -> B.ByteString -> B.ByteString
 signature cr ah input = Base64.encode sig
     where
       sig = case ah of
-              HmacSHA1 -> computeSig SHA1
-              HmacSHA256 -> computeSig SHA256
-      computeSig :: HashAlgorithm a => a -> ByteString
-      computeSig t = toBytes (hmacAlg t (secretAccessKey cr) input)
+              HmacSHA1 -> ByteArray.convert (CMH.hmac (secretAccessKey cr) input :: CMH.HMAC CH.SHA1)
+              HmacSHA256 -> ByteArray.convert (CMH.hmac (secretAccessKey cr) input :: CMH.HMAC CH.SHA256)
```

Comment:
This is the one bit I'm definitely uncertain about, as the cryptohash docs says that the first argument to `hmacAlg` is unused and cryptonite doesn't implement `hmacAlg` at all. I tried to copy what happens in the below case and it compiles, but anything more than that is beyond my minimal Haskell skills.